Add Theatron to Architecture_oM and Humans_oM

rolyhudson opened the pull request:
Theatron is a set of Architecture objects for defining the geometry of stadia, theatre, cinemas, lecture halls and auditoria. It includes Human objects related to users of these spaces: Spectators, Audience, Eye, ViewCones and analytic methods for evaluating view quality and ergonomics

### NOTE: Depends on 
A fork of BHoM_Engine with object definitions in Architecture_Engine and Humans_Engine
pull request in BHoM_Engine: https://github.com/BHoM/BHoM_Engine/pull/1107 
### Test files
Three zipped grasshopper files that demonstrate three key use case scenarios
[theatron-GH_Tests.zip](https://github.com/BHoM/BHoM_Engine/files/3418742/theatron-GH_Tests.zip)

### Additional comments
This is first time pull request to BHoM so its wrong, too complicated or missing something please let me know

rolyhudson commented:
Hi @al-fisher :sunglasses: thanks for reviewing. I have skimmed your thoughts all really helpful stuff :thumbsup: :thumbsup:. Just starting to go through your comments one by one and will send further thoughts.

rolyhudson commented on Architecture_oM/Theatron/Elements/ActivityArea.cs:
```diff
@@ -0,0 +1,45 @@
+
+        public double Length { get; set; } = 0.0;
+
+        /***************************************************/
```
@al-fisher Width and Length properties are obsolete now as they relate only to the football / rectangular pitch stadia use case which is quite a special case. I'll check this remove them an see if there are other implications. I'll also take a look at the ShapeProfiles you mention. I think the Polyline may allow greatest flexibility. In the case of an analysis of a curved cinema screen this is input as a 3d Polyline for the ActivityArea.

rolyhudson commented on Architecture_oM/Theatron/Elements/SeatingBlock.cs:
```diff
@@ -0,0 +1,59 @@
+
+        public Mesh Floor { get; set; } = new Mesh();
+
+        public Audience Audience { get; set; } = new Audience();
```
:thumbsup:

rolyhudson commented on Humans_oM/ViewQuality/AvalueSettings.cs:
```diff
@@ -0,0 +1,45 @@
+
+        public double ForeheadSize { get; set; } = 0.120;
+
+        public double NearHeadRange { get; set; } = 0.100;//dist from viewplane centre for nearest heads
```
@al-fisher Yes that is much needed across the Theatron_oM and related HUmans_oM

rolyhudson commented on Humans_oM/ViewQuality/Spectator.cs:
```diff
@@ -0,0 +1,41 @@
+        /**** Properties                                ****/
+        /***************************************************/
+
+        public BodyParts.Eye Eye { get; set; } = new  BodyParts.Eye();
```
in the land of the blind the one eyed man is king :joy: :joy: :joy:. EyePair plus a ref point is a good idea

al-fisher replied:
Cool - I am really liking the idea of a `PairOfEyes` object now actually! 👀 
Think this approach can be really useful for a `PairOfEars` too in the future for our acoustical friends and binocular sounds. 👂👂

rolyhudson replied:
Looking at this I think the Spectator should have a ```Head``` property and ```Head``` should have the ```PairOfEyes``` property and a 2d Polyline ```HeadOutline```
I'm using ```HeadOutline``` in evaluation of occluded views in Avalue so i think this hierarchy makes sense

rolyhudson replied:
> Looking at this I think the Spectator should have a `Head` property and `Head` should have the `PairOfEyes` property and a 2d Polyline `HeadOutline`
> I'm using `HeadOutline` in evaluation of occluded views in Avalue so i think this hierarchy makes sense

Made this change and fixed Engine related issues

rolyhudson commented:
> I was also looking at the Parameters objects you have created and how they could potentially be broken into either smaller identifiable subcategories. Or assigned as Properties directly to the number of host objects rather than flat list of variables. The idea being to make more modular and increase flexibility of workflow and model generation. 

This is a good point @al-fisher especially with the ``` Theatron ``` objects in ``` Architecture_oM```  and ``` ViewQuality```  in ``` Humans_oM``` . It makes sense to pull out parameters related to eye positions -seated, standing in wheelchair and put those into a  ```EyePositionParameters``` object in ``` ViewQuality```  in ``` Humans_oM```. Other logical modules that I'm splitting out are those related to vomitories and those related to SuperRisers

rolyhudson commented on Architecture_oM/Theatron/Parameters/ProfileParameters.cs:
```diff
@@ -0,0 +1,75 @@
+        public double StandingEyePositionX { get; set; } = 0.0;
+
+        public double RiserHeightRounding { get; set; } = 0.0;
+
```
Broken these parameters into modules
EyePositionParameters (in Humans_oM), VomitoryParameters and SuperRiserParameters

rolyhudson commented:
I've addressed all these suggested changes and committed. Still to add the classic stadium parameters to Datasets. Also updated the GH demos:
I'm not quite sure whats next in the process but let me know if action is needed from me. 👍 👍 👂 👀 👂 

[theatron GH demos.zip](https://github.com/BHoM/BHoM/files/3458810/theatron.GH.demos.zip)
